### PR TITLE
Search(Playground): Sync data when Move and Delete Dashboards

### LINKED ISSUE #48819: Folder Management: Implement new designs on Browse and Search page (Beta 9.0)
**Browse and Search page**

Latest designs in [Figma](https://www.figma.com/file/EfwqyqijsrwqO1D1GfsStn/Go-Git-It?node-id=136%3A9372) - caveats:
No folder nesting in this stage yet
Only changing the results table, the other UI elements (filtering, add new etc.) stay as they are

**Changes to the playground search page** 

#### Folder View
- [x] Implement hierarchy in folder view (@ryantxu)
	- [x] #49042
	- [x] In Folder view: 
		- [x] fix alignment of dashboards when a section is expanded
		- [x] When toggle "Show manage options" is enabled, dashboards do not contain a checkbox
		- [x] When checking folder, children's dashboards are not being checked
		- [x] "Go to folder"  as of right now
		- [x] Implement Recents and Starred
- [x] Implement Delete and Move actions in New Search @axelavargas 
  - [x] Check user permissions
  - [x] Update list of items when moving or deleting items (Frontend), backend is working on making sure the Grafana Data source is keeping the items up to date
  - [x]  Delete should be disabled if the folder is General
  - [x]  Only available in /Browse and Folder page

#### List View

- [x] Add text 'No panel title specified (or something similar)' when we show panels without a title. Check with UX about the font and color that we can use.

#### Filtering and Sorting

- [x] Implement filtering by tags in Search Table
- [x] Implement filtering by tags in Folder View
- [x] Sorting
	- [x] Verify we get extra sorting options in enterprise
    - [x] Add sorting feature (backend and frontend) 
 
#### A11y and Styling
- [x]  Fix search table to be keyboard accessible @kaydelaney 
- [x]  Fix animation for table when responsive @kaydelaney 
- [x]  Fix height in table @kaydelaney 
- [x]  Throttle + debounce query execution (currently every keypress) @kaydelaney 
- [x]  Verify grid view is only enabled when feature flag is present
- [x] Fix Table Style/Css, we are now using `Cell` instead of 'DefaultCell' @axelavargas 

#### Performance

- [x] Table search has performance issues when searching, the table is being rendered multiple times for every keypress @kaydelaney

### PR DESCRIPTION
**Pull request by axelavargas**
**What this PR does / why we need it**:

When moving and Deleting dashboards, we need  to keep the list of items in the search table updated.

Approach in this PR:
- Remove old `onMoveItems` and `onDeleteItems` functions
- Pass callback when user move or delete items
- Add tiny improvement to show spinner when moving items
- Clear up the selected item's list 
- Trigger again the request to the backend (🤔 I am not sure this is the right way) 

**Which issue(s) this PR fixes**:

Related #48819

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/search/page/SearchPage.tsx [hunk 1]
```diff
@@ -88,6 +88,14 @@ export default function SearchPage() {
     setSearchSelection(updateSearchSelection(searchSelection, !current, kind, [uid]));
   };
 
+  const onUpdateList = async () => {
+    // clean up search selection
+    setSearchSelection(clearSearchSelection(searchSelection));
+    // trigger again the search to the backend
+    // TODO: the backend is not returning the updated value :(
+    onQueryChange(inputValue);
```
Comment (kaydelaney) [inline public/app/features/search/page/SearchPage.tsx:96]: I think the backend might need to re-index or something (I say this but I don't really know how bluge works or anything). Restarting grafana-server fixes it for me

File: public/app/features/search/page/SearchPage.tsx [hunk 2]
```diff
@@ -88,6 +88,14 @@ export default function SearchPage() {
     setSearchSelection(updateSearchSelection(searchSelection, !current, kind, [uid]));
   };
 
+  const onUpdateList = async () => {
+    // clean up search selection
+    setSearchSelection(clearSearchSelection(searchSelection));
```
Comment (kaydelaney) [inline public/app/features/search/page/SearchPage.tsx:93]:
```suggestion
    setSearchSelection(newSearchSelection());
```

Maybe we could just reuse newSearchSelection here?

**Review by kaydelaney — APPROVED** [review]

**Comment by kaydelaney** [comment]
Looks good! Just some small comments/suggestions 🙂

File: public/app/features/search/page/SearchPage.tsx (hunk 1 above)
Comment (axelavargas) [reply public/app/features/search/page/SearchPage.tsx:96]: Yes, I asked and the issue is known and being worked on :)

**Pull request merged**